mantidproject/mantid
# HistogramData module (X data only)

**Do not merge this PR, this is still in an early review stage.**

**Edit: Work has progressed, please see https://github.com/mantidproject/mantid/pull/16627 instead of this PR.**

Note that the documentation is still quite incomplete. I will work on this during the review process.

There is a `rst` file in `concepts` that tries to explain some details, it might be worth having a look at it before the review.

**Code review:**
This is a very large and critical change, I would like to suggest to split the review into several parts:
1. Review everything in the new module `HistogramData`. **<- current phase**
2. Feedback to me, I will then review the full set of changes myself.
3. Review of everything else (existing Mantid code).

I think we should have more than one reviewer for both 1) and 3).

**To test:**

This is a global change, there is no individual test. If you have access to any Python scripts using Mantid that are not part of our automated testing it would be good to check if they still work.

Fixes #16086.



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
